markup.js

```diff
@@ -271,4 +203,30 @@ function buildKeyboard (buttons, options) {
   return result
 }
 
+function tag (name, params) {
+  return {
+    open: params
+      ? `<${name} ${Object.entries(params).map(([key, value]) => `${key}="${value}"`).join(' ')}>`
+      : `<${name}>`,
+    close: `</${name}>`
+  }
+}
+
+const HTMLTags = new Map([
+  ['bold', tag('b')],
+  ['italic', tag('i')],
+  ['code', tag('code')],
+  ['pre', tag('pre')],
+  ['strikethrough', tag('s')],
+  ['underline', tag('u')],
+  ['text_link', ({ url }) => tag('a', { href: url })],
+  ['text_mention', ({ user }) => tag('a', { href: `tg://user?id=${user.id}` })]
+])
+
+function getHTMLTag (entity) {
+  const tag = HTMLTags.get(entity.type || 'unknown')
+  if (!tag) throw new TypeError('Unknown entity')
```

Comment:
This will throw if [`entity.type`] is “mention”, “hashtag”, “cashtag”, “bot_command”, “url”, “email” or “phone_number”.
```suggestion
  if (!tag) return { open: '', close: '' }
```

[`entity.type`]: https://core.telegram.org/bots/api#messageentity